src/plots/cartesian/layout_attributes.js

```diff
@@ -1190,6 +1190,28 @@ module.exports = {
             'Used with `categoryorder`.'
         ].join(' ')
     },
+    categoryshapeshiftstart: {
+        valType: 'number',
+        dflt: 0,
+        min: -0.5,
+        max: 0.5,
```

Comment:
Perhaps increasing the range of `min` and `max` a bit could be useful to highlight points for cases e.g. histogram and bars with tiny gaps.
I suggest we allow values between -0.7 and 0.7 or maybe even -1 and 1?